### PR TITLE
Added attributes to the Config

### PR DESCRIPTION
**Pull request by hamidadelyar**
Hey, 

Really appreciate this package, great work! 
Made a small tweak to allow attributes for the <style> tag rather than assuming type="text/css"

Let me know what you think!

Thanks, 

Hamid

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,15 +66,20 @@ interface Config {
     position?: 'before' | 'after'
     removeTarget?: boolean
   }
+  attributes?: any
```
Comment (Runjuu) [inline README.md]: How about

```ts
attributes?: { [name: string]: string }
```

File: src/core/base-plugin.ts [hunk 2]
```diff
@@ -75,7 +75,26 @@ export class BasePlugin {
     htmlFileName: string
     style: string
   }) {
-    const styleString = `<style type="text/css">${style}</style>`
+    let attributesString = ''
+
+    if (this.config.attributes) {
+      if (
+        typeof this.config.attributes !== 'object' &&
+        this.config.attributes === null
```
Comment (Runjuu) [inline src/core/base-plugin.ts]: Seems `if (this.config.attributes)` already exclude `null`.

How about:

```ts
if (this.config.attributes && typeof this.config.attributes === 'object') {
  ...
}
```

File: src/core/base-plugin.ts [hunk 3]
```diff
@@ -75,7 +75,26 @@ export class BasePlugin {
...
+      ) {
+        throw new Error(
+          `Please provide a key/value object if intending to use the attributes option`,
+        )
+      }
+
+      Object.keys(this.config.attributes).map((key) => {
+        const value = this.config.attributes[key] || ''
+        attributesString += ` ${key}="${value}"`
+      })
```
Comment (Runjuu) [inline src/core/base-plugin.ts]: `map` is not a good way to do `side effect` things, use `forEach` instead.

Comment (Runjuu) [reply src/core/base-plugin.ts]: A better way to get `attributesString` is extract a function to do this. Then we could avoid creating a temp variable.

```ts
function getAttributesString(config: Config): string {
  if (config.attributes && typeof config.attributes === 'object') {
    return Object
      .keys(config.attributes)
      .map((key) => `${key}="${config.attributes[key]}"`)
      .join(' ')
 }

  if (config.attributes !== undefined && config.attributes !== null) {
    throw new Error(
      `Please provide a key/value object if intending to use the attributes option, not ${config.attributes}`,
    )
  }
 
  return ''
}

protected addStyle() {
  ...
  const styleString = `<style ${getAttributesString(this.config)}>${style}</style>`
  ...
}
```

File: src/types.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ export interface Config {
   filter?(fileName: string): boolean
   leaveCSSFile?: boolean
   replace?: ReplaceConfig
+  attributes?: any
```
Comment (Runjuu) [inline src/types.ts]: Same as above.

**Comment by Runjuu** [comment]
Hi @hamidadelyar ,I realized that there is a problem with `object` type of `attributes`. We can't add `attributes` for specific file. How about change the signature to

```ts
{
  attributes?(fileName: string): { [name: string]: string }
}
```

**Comment by hamidadelyar** [comment]
> Hi @hamidadelyar ,I realized that there is a problem with `object` type of `attributes`. We can't add `attributes` for specific file. How about change the signature to
> 
> ```ts
> {
>   attributes?(fileName: string): { [name: string]: string }
> }
> ```

**Comment by hamidadelyar** [comment]
> Hi @hamidadelyar ,I realized that there is a problem with `object` type of `attributes`. We can't add `attributes` for specific file. How about change the signature to
> 
> ```ts
> {
>   attributes?(fileName: string): { [name: string]: string }
> }
> ```

**Comment by hamidadelyar** [comment]
> Hi @hamidadelyar ,I realized that there is a problem with `object` type of `attributes`. We can't add `attributes` for specific file. How about change the signature to
> 
> ```ts
> {
>   attributes?(fileName: string): { [name: string]: string }
> }
> ```

Hey @Runjuu,

I'm not entirely sure what you mean. Do you mean changing the Config in types.ts to this?

```ts
export interface Config {
  filter?(fileName: string): boolean
  leaveCSSFile?: boolean
  replace?: ReplaceConfig
  attributes?(fileName: string): { [name: string]: string }
}
```

I'm not entirely sure what this does, also typescript complains about it.

Feel free to make the change :)

**Comment by Runjuu** [comment]
Hi @hamidadelyar , I will merge the PR into `feat/attributes` branch first and make some changes before release.

**Pull request merged**